generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java

```diff
@@ -96,7 +96,13 @@ public void delete(<%= pkType %> id) {
      *  @return the list of entities
      */<% if (databaseType == 'sql') { %>
     @Transactional(readOnly = true) <% } %>
-    public List<<%= instanceType %>> search(String query) {
-        <%- include('../../common/search_template', {viaService: viaService}); -%>
+    public <% if (pagination != 'no') { %>Page<<%= entityClass %><% } else { %>List<<%= instanceType %><% } %>> search(String query<% if (pagination != 'no') { %>, Pageable pageable<% } %>) {
```

Comment:
I followed the same pattern as the loadAll service. The template now contains the method signature for the resource search method (like loadAll) so it would not work.

I just noticed the formatting is off however.